File: db/migrate/0004_create_comments.rb
# -*- coding: utf-8 -*-

class CreateComments < ActiveRecord::Migration
  def change
    create_table(:comments) do |t|
      ## Database authenticatable
      t.boolean :visible, null: false

      t.string :author, null: false
      t.string :email
      t.string :url

      t.json :attrs, null: false, default: "{}"

      t.text :content, null: false

      t.timestamps
    end

    add_reference :comments, :post, index: true, null: false
  end
end

# eof

File: priv/repo/migrations/0004_create_comments.exs
defmodule Burp.Repo.Migrations.CreateComments do
  use Ecto.Migration

  def change do
    create table(:comments) do
      add :visible, :boolean, default: false, null: false
      add :author, :string, null: false
      add :email, :string
      add :url, :string
      add :attrs, :map, null: false, default: %{}
      add :content, :text, null: false

      add :post_id, references(:posts, on_delete: :delete_all, on_update: :update_all)

      timestamps()
    end

    create index(:comments, [:post_id])
  end
end
